### PR TITLE
Update Bitwuzla code for new API

### PR DESCRIPTION
**Pull request by samanthaarcher0**
Add support for the new C++ API of Bitwuzla

### CODE REVIEW

File: bitwuzla/src/bitwuzla_solver.cpp [hunk 1]
```diff
@@ -199,17 +192,21 @@ UnorderedTermMap BzlaSolver::get_array_values(const Term & arr,
...
-  for (size_t i = 0; i < size; ++i)
+  try
+  {
+    std::vector<bitwuzla::Term> bcore = get_bitwuzla()->get_unsat_assumptions();
+    for (auto && elt : bcore)
+    {
+      out.insert(std::make_shared<BzlaTerm>(elt));
+    }
+  }
+  catch (std::exception & e)
```
Comment (CyanoKobalamyne) [inline bitwuzla/src/bitwuzla_solver.cpp:203]: @samanthaarcher0 why did you wrap this in a try-catch?

File: tests/CMakeLists.txt.in [hunk 2]
```diff
@@ -6,7 +6,7 @@ include(ExternalProject)
 ExternalProject_Add(googletest
   GIT_REPOSITORY    https://github.com/google/googletest.git
   # Temporary: to avoid https://github.com/google/googletest/issues/2711
-  GIT_TAG           release-1.12.1
+  GIT_TAG           v1.14.0
```
Comment (CyanoKobalamyne) [inline tests/CMakeLists.txt.in:9]: Ah we should probably put this in a separate PR.

File: bitwuzla/src/bitwuzla_solver.cpp (hunk 1 above)
Comment (CyanoKobalamyne) [reply bitwuzla/src/bitwuzla_solver.cpp:205]: Ah wait I see now, we are wrapping internal solver errors in our own `InternalSolverException`. Maybe we should do this everywhere.

**Comment by CyanoKobalamyne** [comment]
@yoni206 I think this is good to go now, please take a look when you have the chance.

**Review by yoni206 — COMMENTED** [review]
Thanks for all of this. It looks fine, though I am wondering if this would be a good opportunity to add a corresponding test directory to here https://github.com/stanford-centaur/smt-switch/tree/main/tests (I think bitwuzla is the only solver that doesn't have one, right?)

**Comment by CyanoKobalamyne** [comment]
I can do that, but this PR is already big enough, so maybe let's merge that separately.

**Comment by CyanoKobalamyne** [comment]
Actually, what are the solver-specific tests testing that the top-level ones aren't? They mostly just seem a lot older.

**Review by yoni206 — APPROVED** [review]

**Comment by yoni206** [comment]
You are right, it is not really related to this PR. Merged.
Also, I actually don't see a real difference between these two types of tests.
For example, the tests inside the solver directories also just create a solver with the factory, nothing more internal than that.
So perhaps we can at some point port these tests into the unit tests. But that's also not related to this PR and is not urgent.

Thanks for updating bitwuzla!

**Comment by CyanoKobalamyne** [comment]
Yay, of course, thanks for reviewing! And thank you to @samanthaarcher0! :)
